draco/synthesis/noise.py

```diff
@@ -115,16 +132,28 @@ def process(self, data):
         df = data.index_map["freq"]["width"][0] * 1e6
         nfreq = data.vis.local_shape[0]
         nprod = len(data.index_map["prod"])
+        ninput = len(data.index_map["input"])
+
+        # Consider if this data is stacked over redundant baselines or not.
+        if (self.telescope is not None) and (nprod == self.telescope.nbase):
+            redundancy = self.telescope.redundancy
+        elif nprod == nfeed * (nfeed + 1) / 2:
+            redundancy = np.ones(nprod)
+        else:
+            raise ValueError("Unexpected number of products")
 
-        # Calculate the number of samples
-        nsamp = int(self.ndays * dt * df)
+        # Calculate the number of samples, this is a 1D array for the prod axis.
+        nsamp = int(self.ndays * dt * df) * redundancy
         std = self.recv_temp / np.sqrt(2 * nsamp)
 
         with mpi_random_seed(self.seed):
-            noise_real = std * np.random.standard_normal((nfreq, nprod, ntime))
-            noise_imag = std * np.random.standard_normal((nfreq, nprod, ntime))
+            noise_real = std[np.newaxis, :, np.newaxis] * np.random.standard_normal(
```

Comment:
In here you could just use `cora.util.nputil.complex_std_normal` to save a couple of lines. You'd need to remove the factor of 2 in `std` and when you set the weights though.

Reply:
Ok this should be done.